### PR TITLE
More GSOD improvements

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
Hey @efedotova -> I made a first quick attempt at separating TESTING.rst from Local/Breeze. I hope it might be a good start for you as discussed :).

File: TESTING.rst [hunk 1]
```diff
@@ -0,0 +1,712 @@
...
+---------------------------------
+
+To run all tests with default settings (Python 3.6, Sqlite backend, "docker" environment), enter:
+
+.. code-block::
+
+  ./scripts/ci/local_ci_run_airflow_testing.sh
+
+
+To select Python 3.6 version, Postgres backend, and a ``docker`` environment, specify:
```
Comment (efedotova) [inline TESTING.rst:197]: does it mean "edit the .sh script"? or is this an input for the script we launched?

File: LOCAL_VIRTUALENV.rst [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
 
 Local Virtual Environment (virtualenv)
-============================================
+======================================
 
-Use the local virtualenv development option in the combination with the _`Breeze <BREEZE.rst#aout-airflow-breeze>`_
+Use the local virtualenv development option in the combination with the `Breeze <BREEZE.rst#aout-airflow-breeze>`_
 development environment. This option helps you benefit from the infrastructure provided
-by your IDE (for example, IntelliJ's PyCharm/Idea) and work in the enviroment where all necessary dependencies and tests are 
+by your IDE (for example, IntelliJ's PyCharm/Idea) and work in the enviroment where all necessary dependencies and tests are
```
Comment (turbaszek) [inline LOCAL_VIRTUALENV.rst]:
```suggestion
by your IDE (for example, IntelliJ Idea / PyCharm) and work in the enviroment where all necessary dependencies and tests are
```
Those are two different products of JetBrains.

Comment (turbaszek) [inline LOCAL_VIRTUALENV.rst]:
```suggestion
by your IDE (for example, IntelliJ's PyCharm/Idea) and work in the environment where all necessary dependencies and tests are
```

**Review by turbaszek — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -267,348 +264,34 @@ Limitations:
...
-Airflow test suite is based on Travis CI framework as running all of the tests
-locally requires significant setup. You can set up Travis CI in your fork of
-Airflow by following the `Travis
-CI Getting Started
-guide <https://docs.travis-ci.com/user/getting-started/>`__.
+Static code checks
+==================
 
+We check our code quality via static code checks. See
+`STATIC_CODE_CHECKS.rst <STATIC_CODE_CHECKS.rst>`_ for details.
```
Comment (ashb) [inline CONTRIBUTING.rst]:
```suggestion
`STATIC_CODE_CHECKS.rst`_ for details.
```

works I think?

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -267,348 +264,34 @@ Limitations:
...
-    organization; otherwise, click the **Request** button. For the Travis CI
-    Authorized OAuth App, you may have to grant access to the forked
-    ``ORGANIZATION/airflow`` repo even though it is public.
-
-3.  Access Travis CI for your fork at
-    `<https://travis-ci.org/ORGANIZATION/airflow>`_.
-
-Creating New Projects in Travis CI
-----------------------------------
+* **Unit tests** are Python ``nose`` tests launched with ``run-tests``.
```
Comment (ashb) [inline CONTRIBUTING.rst:282]: pytest now? Or have we not quite merged that yet @nuclearpinguin ?

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -742,8 +495,56 @@ Resources & links
```
Comment (ashb) [inline CONTRIBUTING.rst:498]: Can't comment on the line here, but we the all-commits list is high volume so it's probably not recommended for most contributors to be subscribed to it. Even the dev list is optional for them.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -742,8 +495,56 @@ Resources & links
...
+     `pre-commits installed <STATIC_CODE_CHECKS.rst#pre-commit-hooks>`__,
+     this step is automatically run while you are committing your code. If not, you can do it manually
+     via ``git add`` and then ``pre-commit run``.
+
+2. Rebase your fork, squash commits, and resolve all conflicts.
+
+3. Re-run static code checks again.
+
+4. Create a pull request with the following title for the sample ticket:
+   [AIRFLOW-5934: Added extra CC: field to the Airflow emails].
```
Comment (ashb) [inline CONTRIBUTING.rst]:
```suggestion
   ``[AIRFLOW-5934] Added extra CC: field to the Airflow emails.``
```

We should make a mention here about commit messages - that they should be short, but descriptive enough  to be useful in a changelog.

**Review by ashb — COMMENTED** [review]
I wonder if the images should live under `docs/images/` instead of just `images/`?

**Comment by potiuk** [comment]
> I wonder if the images should live under docs/images/ instead of just images/?

I think those documents are not part of the user docs - they are developer's DOC's. They never end up in the airflow sphinx documentation so I would not mix them. We can change it to something else - but then the whole CONTRIBUTING.rst should also be there I think.

Another option might be that we actually MOVE the contributing documents to the docs/folder and make it part of the documentation under the "contributing" section. However I feel that contributors should use github rendering capabilities rather than generated docs.

WDYT?

**Comment by ashb** [comment]
Yeah, I think keeping it in root has advantages for Github, (including a notice when it changes) which would be lost if we move the whole thing to live under docs/

Maybe we could do a `.. include:` or some mechanism to have the contributing guide etc in the build docs? Alternatively maybe they don't belong in the versioned docs, but just on the website instead?

File: CONTRIBUTING.rst (hunk 4 above)
Comment (turbaszek) [reply CONTRIBUTING.rst:282]: #6472 is not yet merged but yes we will have to make it consistent

Comment (potiuk) [reply CONTRIBUTING.rst:282]: Not yet. We will merge it afterwards.

File: CONTRIBUTING.rst (hunk 5 above)
Comment (potiuk) [reply CONTRIBUTING.rst:498]: Added info on that

File: CONTRIBUTING.rst (hunk 6 above)
Comment (potiuk) [reply CONTRIBUTING.rst]: We have this further down the list: 

> Adhere to guidelines for commit messages described in this `article <http://chris.beams.io/posts/git-commit/>`__.
>     This makes the lives of those who come after you a lot easier.

Comment (ashb) [reply CONTRIBUTING.rst]: Not in this document is it? Oh earlier on, but just not changed so didn't see it in the diff.

Also the suggestion is for correct balancing of `[]` :)

**Comment by potiuk** [comment]
I think it's best to simply refer to CONTRIBUTING.rst in github from the Airflow Website :)

**Comment by potiuk** [comment]
Review comments applied.

File: CONTRIBUTING.rst (hunk 6 above)
Comment (potiuk) [reply CONTRIBUTING.rst]: Right :)

Comment (potiuk) [reply CONTRIBUTING.rst]: @ashb:  It is in this document but in the lines that have not changed :). It's been there since I remember.

**Comment by potiuk** [comment]
Rebased and I think ready to merge if we gen an approval.

**Review by ashb — APPROVED** [review]

**Pull request merged**